sqlglot/parser.py

```diff
@@ -1005,7 +1006,12 @@ def _parse_table_alias(self):
         )
 
     def _parse_subquery(self, this):
-        return self.expression(exp.Subquery, this=this, alias=self._parse_table_alias())
```

Comment:
this would be cleaner if you modified _parse_pivots to return the array

return self.expression(exp.Subquery, this=this, pivots=self._parse_pivots(), alias=self._parse_table_alias())